cnd/src/network/comit_ln.rs

```diff
@@ -105,30 +105,31 @@ impl ComitLN {
         &mut self,
         id: LocalSwapId,
         create_swap_params: HanEtherereumHalightBitcoinCreateSwapParams,
-    ) {
+    ) -> anyhow::Result<()> {
         let digest = create_swap_params.clone().digest();
 
+        if self.swaps_waiting_for_announcement.contains_key(&digest) {
```

Comment:
This is going to result in a 500 for the user because we cannot identify this particular case.

Please create a proper error and send a meaningful error (400) back to the user. An e2e test would also be nice for this case.

I would suggest a single struct (not enum) that derives `thiserror::Error`. You can then do:

```suggestion
        if self.swaps_waiting_for_announcement.contains_key(&digest) {
            anyhow::bail!(TheErrorYouCreated)
        }
```

(I don't think we need to share the digest with the user as it is an internal thing, so I omitted it from the struct ctor)

Reply:
Oh man, that macro is just what I have been looking for.  I have typed `return Err(anyhow!(Error::Foo))` so many times this last week.

Reply:
RTFM man :)

Reply:
> This is going to result in a 500 for the user because we cannot identify this particular case.

Thanks, I did not even begin to think all the way up to the user, good reviewing.